lib/decompress.ps1

```diff
@@ -123,8 +123,11 @@ function Expand-7zipArchive {
     }
     if (!$IsTar -and $ExtractDir) {
         movedir "$DestinationPath\$ExtractDir" $DestinationPath | Out-Null
-        # Remove temporary directory
-        Remove-Item "$DestinationPath\$($ExtractDir -replace '[\\/].*')" -Recurse -Force -ErrorAction Ignore
+        # Remove temporary directory if empty
+        $ExtractDirFullPath = [string] "$DestinationPath\$($ExtractDir -replace '[\\/].*')"
+        if ((Get-ChildItem -Path $ExtractDirFullPath -Force).Count -le 0) {
+            Remove-Item -Path $ExtractDirFullPath -Recurse -Force -ErrorAction Ignore
+        }
```

Comment:
```suggestion
        if ((Get-ChildItem "$DestinationPath\$ExtractDir" -Force -ErrorAction Ignore).Count -eq 0) {
            Remove-Item "$DestinationPath\$($ExtractDir -replace '[\\/].*')" -Recurse -Force -ErrorAction Ignore
        }
```

This is what I mean.

Reply:
Since `extract_dir` could be a subdirectory such as `aaa\bbb`, `Expand-xxx` will delete `aaa` after extraction. Other functions that use a rarely used temporary directory may not encounter this bug, in my opinion.

Reply:
`-le` (less than or equal) will return true if count is 0 too, but sure, I can change that.

I see you also removed `-Path`, why?